src/api/app/policies/report_policy.rb

```diff
@@ -0,0 +1,28 @@
+class ReportPolicy < ApplicationPolicy
+  attr_reader :user, :record
+
+  def initialize(user, record)
+    super(user, record)
+
+    @user = user
+    @record = record
+  end
+
+  def show?
+    user.is_admin? || record.user == user
+  end
+
+  def submit_report?
+    not_self = if record.instance_of?(::Comment)
+                 record.user != user
+               elsif record.instance_of?(::User)
+                 record != user
+               elsif record.instance_of?(::Package)
+                 record.project.users.exclude?(user)
+               elsif record.instance_of?(::Project)
+                 record.users.exclude?(user)
+               end
```

Comment:
💭  I also think you should move this into it's own private method...

Reply:
Just to check inclusion in maintainers? This is just a one-liner and not used anywhere else.  I've updated this method and there are new changes. Please have a look again

Reply:
What do you mean by "just"? Instead of 

```
 not_self = if record.instance_of?(::Comment)
                 record.user != user
               elsif record.instance_of?(::User)
                 record != user
               end
...
``` 

just create a private method by that name. Much more readable.

```
private

def not_self
  if record.instance_of?(::Comment)
    return record.user != user
  elsif record.instance_of?(::User)
    record != user
  end
end
```

which is much more readable IMHO

Reply:
We don't need it anymore. This code has been replaced by other policy classes